### ISSUE TITLE
[poll] How should workspaces work?

### ISSUE THREAD

**Issue by ammen99** [WayfireWM/wayfire#977]
For Wayfire 0.8, I had planned #661 . However, I have started thinking that adding support for both independent workspaces and "viewports" would mean a lot of additional work. Hence, I want to ask the (potential) users of Wayfire what your preference is, to gauge whether making this an option makes sense at all. Here is a quick summary of the two possible modes of operation regarding workspaces:

- Viewports: this is how workspaces in Wayfire work right now. You can have a window standing in between 2 workspaces, and it is visible on both of them. In a certain sense, there is just one big surface, and each "workspace" is just a part of this surface.
- Independent workspaces: this is how workspaces work in GNOME, Sway and many others. A window cannot be visible on two workspaces (except sticky windows).

Of course, if the decision is to move Wayfire towards the independent workspaces camp, all effects for multiple workspaces like vswitch, expo, vswipe and others, will continue working - the reorganization is mostly how workspaces are represented internally.

So, I ask everyone to vote by leaving a reaction to the two comments below, indicating whether you would prefer viewports or independent workspaces. I can't promise everyone will be happy but I want to get a picture of what you guys want. Feel free to also share your thoughts on the matter :)

P.S. This issue will be open until the 0.7 release.

**Comment by ammen99** [comment]
I would prefer viewports.

**Comment by ammen99** [comment]
I would prefer true workspaces.

**Comment by ammen99** [comment]
Some important observations:

- as @soreau said on IRC, true workspaces would also mean that views will be cut off at workspace boundaries in vswitch/vswipe, expo, cube, etc. In the special case of Cube, we could implement it in a different way which avoids the problem, but for the other plugins, the views will be cut off.
- Having true workspaces would, on the other hand, bring some benefits when trying to fix #582 and #970, and maybe allow better Xwayland integration.

**Comment by ammen99** [comment]
And one more thing: retaining the current model (viewports) also means that existing bugs will, of course,be fixed (like #970)

**Comment by damianatorrpm** [comment]
As noted on irc I hope I don't spam the poll, I want to clarify:
If the xwayland bugs are fixed, I don't care either way,
but I care that the bugs are fixed.

If 'true workspaces' would provide all (without exception from what @soreau listed on irc) things the current implementation
I would slightly favor it (but I imagine it's a lot of work and may not be worth it) for the reason that it is kinda strange
that I get the view-workspace-changed on what I would call normal space i.e moving the window just slightly to the right/bottom etc), but I think that signal happens only once when a view finishes moving which could be tweaked as well.

**Comment by marcusbritanicus** [comment]
I prefer to have true workspaces.

**Comment by chaoskagami** [comment]
Can I pick the third option of awesome/sway style tags?

Joking aside...if I have to pick, I'm partial to workspaces. Functionally, having half the window visible on viewports is pointless aside from looking pretty and can actually be counterproductive if it overlaps something else. If switching to workspaces fixes bugs with positioning and z-ordering, I say go for it.

**Comment by travankor** [comment]
Crazy idea but... I wonder if we could have a hybrid option that is like viewports but lets you add a little padding/margin between viewports.

**Comment by ammen99** [comment]
> Crazy idea but... I wonder if we could have a hybrid option that is like viewports but lets you add a little padding/margin between viewports.

We could have ... But it will make stuff quite confusing in Expo :D And also in other places ...

**Comment by travankor** [comment]
It could be optional, basically like i3/sway gaps :p

This way a view can't end up on a different workspace when a small part is outside the current workspace.

**Comment by ammen99** [comment]
Initially, I thought that many more people want to have truly independent workspaces - however, based on the responses, it seems that this is not the case. Providing truly independent workspaces will also be a lot of work, potentially many breaking changes for plugins, and will inevitably introduce new bugs. This is why my decision is to postpone that feature for the long-term future, probably post 1.0. Instead, on the next release I'll focus on improving the current workspace implementation (like the already open PRs #1063, #1064).

Thanks to everyone for your feedback!
